src/appleseed.studio/mainwindow/project/projectmanager.cpp

```diff
@@ -89,8 +91,14 @@ void ProjectManager::load_project_async(const string& filepath)
 
     m_async_io_filepath = QString::fromStdString(filepath);
 
+#if defined(__GNUC__) && (__GNUC__ > 4)
     m_async_io_future_watcher.setFuture(
         QtConcurrent::run(this, &ProjectManager::do_load_project, filepath));
+#else
+    const bool successful = do_load_project(filepath);
```

Comment:
Is this actually working? An alternative would be to spawn a thread, it's basically what QtConcurrent does except that it picks the thread from a thread pool.

Reply:
In my tests it worked. It is doing the same thing as before but on the main thread, no?

Reply:
Ok.

> It is doing the same thing as before but on the main thread, no?

Yes. I guess the only difference is that log messages emitted during loading won't be visible until the project is actually fully loaded (or has failed to load).